Makefile

```diff
@@ -1,70 +1,29 @@
-ALERTING_RULE_FILES = rhobs/alerting/*.yaml
+SHELL=/usr/bin/env bash -o pipefail
 
-.PHONY: all
-all: prepare sync_pipenv lint test_rules pint_lint lint_yamls kustomize_build
-
-.PHONY: prepare
-prepare: pint promtool yq kustomize
-	echo "Extract Prometheus rules"
-	(./yq eval-all '. as $$item ireduce ({}; . *+ $$item)' ${ALERTING_RULE_FILES} | ./yq ".spec" ) > ${EXTRACTED_RULE_FILE}
+VPATH ?= $(PWD):$(PATH)
+# Runtime CLI to use for running images
+CONTAINER_ENGINE ?= $(shell command -v podman 2>/dev/null || echo docker)
+IMG_VERSION=1.0.4
 
-.PHONY: lint
-lint:
-	echo "Running Prometheus rules linter tests"
-	./promtool check rules ${EXTRACTED_RULE_FILE}
+BASE_CMD=$(CONTAINER_ENGINE) run -v "$(shell pwd):/work" --privileged -t quay.io/rhobs/obsctl-reloader-rules-checker:$(IMG_VERSION) -t rhtap -d rhobs/alerting -y -p
 
-.PHONY: test_rules
-test_rules:
-	echo "Running Prometheus rules unit tests"
-	./promtool test rules test/promql/tests/*
-
-pint:
-	echo 'Installing pint...'
-	curl -OJL 'https://github.com/cloudflare/pint/releases/download/v${PINT_VERSION}/pint-${PINT_VERSION}-linux-amd64.tar.gz'
-	tar xvzf 'pint-${PINT_VERSION}-linux-amd64.tar.gz' pint-linux-amd64
-	mv pint-linux-amd64 pint
-	rm 'pint-${PINT_VERSION}-linux-amd64.tar.gz'	
+.PHONY: all
+all: check-and-test kustomize-build
 
-promtool:
-	echo "Installing promtool..."
-	curl -OJL "https://github.com/prometheus/prometheus/releases/download/v${PROMTOOL_VERSION}/prometheus-${PROMTOOL_VERSION}.linux-amd64.tar.gz"
-	tar xvzf "prometheus-${PROMTOOL_VERSION}.linux-amd64.tar.gz" "prometheus-${PROMTOOL_VERSION}.linux-amd64"/promtool --strip-components=1
-	rm -f prometheus-${PROMTOOL_VERSION}.linux-amd64.tar.gz
+.PHONY: check-and-test
+check-and-test:
+	$(BASE_CMD) --tests-dir test/promql/tests
 
-yq:
-	echo "Installing yq..."
-	curl -OJL "https://github.com/mikefarah/yq/releases/download/v${YQ_VERSION}/yq_linux_amd64.tar.gz"
-	tar xvzf yq_linux_amd64.tar.gz ./yq_linux_amd64
-	rm -f yq_linux_amd64.tar.gz
-	mv yq_linux_amd64 yq
+.PHONY: check
+check:
+	$(BASE_CMD)
 
 kustomize:
 	curl -s "https://raw.githubusercontent.com/kubernetes-sigs/kustomize/master/hack/install_kustomize.sh"  | bash
 
-.PHONY: pint_lint
-pint_lint:
-	echo "Linting Prometheus rules..."
-	./pint --no-color lint ${EXTRACTED_RULE_FILE}
-
-.PHONY: install_pipenv
-install_pipenv:
-	python3 -m pip install pipenv
-
-.PHONY: sync_pipenv
-sync_pipenv:
-	python3 -m pipenv sync --dev
-
-.PHONY: lint_yamls
-lint_yamls:
-	python3 -m pipenv run yamllint . && echo "lint_yamls: SUCCESS"
-
-.PHONY: kustomize_build
-kustomize_build:
+.PHONY: kustomize-build
+kustomize-build: kustomize
 	# This validates that the build command passes and not its output's validity.
 	# It will fail once we have more than one subdirectory, which will prevent us from
 	# adding untested configurations (this target will have to chage when that happens).
-	./kustomize build prometheus/* 1>/dev/null
+	kustomize build prometheus/* 1>/dev/null
```

Comment:
I am now defining the `VPATH` variable which is made of the current working dir (`$PWD`) and the `PATH` variable.
This `VPATH` variable is used by `make` to locate the binaries it needs to run or compute.

So there are now 2 possibilities:
- The user already installed `kustomize` on his system: in that case the `Makefile` won't install it (i.e. `kustomize` will do nothing) but it will still use it in the `kustomize-build` target.
- The user did not install `kustomize` on his system: in that case the binary will be retrieved through the `kustomize` directory in the current working dir and it will be used later in the `kustomize-build` target.

Reply:
Alright. Learning something new every day I suppose :)

Regardless, the regression that we had with testing is due to a change we did that makes this kustomize test redundant.

All in all, we're going to drop it this sprint or the next so if it gives you trouble you can apply whatever change you need for stopping it from running so it's not a stopper for you..